core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -2592,13 +2594,22 @@ private Type coerceToSingleType(StackableAstVisitorContext<Context> context, Str
 
         private void addOrReplaceExpressionCoercion(Expression expression, Type type, Type superType)
         {
-            NodeRef<Expression> ref = NodeRef.of(expression);
-            expressionCoercions.put(ref, superType);
+            addOrReplaceExpressionsCoercion(List.of(expression), type, superType);
```

Comment:
"batch" version of `addOrReplaceExpressionsCoercion` looks complex and unnecessary

i would rather keep the old method `addOrReplaceExpressionCoercion(Expression expression, Type type, Type superType)` as is, and just call it in the loop in the only place where you would call it with many expressions.

Am i missing anything?

Reply:
I wanted to reduce the number of `typeCoercion.isTypeOnlyCoercion(type, superType)` calls, hence the batching.

Reply:
Then maybe overload, allowing the caller to provide this information?